test/infrastructure/inmemory/pkg/server/api/watch.go

```diff
@@ -108,6 +109,35 @@ func (h *apiServerHandler) watchForResource(req *restful.Request, resp *restful.
 		return err
 	}
 
+	if resourceVersion != "" {
```

Comment:
I only see one problem and that is that this gets deadlocked if we have more than 1k objects (because the event chan has size 1000)

Or potentially earlier if this gets called multiple times before the channel is emptied in between.

WDYT about moving l.124-138 into a go routine?

Reply:
changed to not handle the catchup via the event channel.

This way we don't get a deadlock and ensure the events get written in order.